pkg/dao/rpm.go

```diff
@@ -0,0 +1,171 @@
+package dao
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"gorm.io/gorm"
+)
+
+type rpmDaoImpl struct {
+	db *gorm.DB
+}
+
+func GetRpmDao(db *gorm.DB) RpmDao {
+	return rpmDaoImpl{
+		db: db,
+	}
+}
+
+func (r rpmDaoImpl) isOwnedRepository(orgID string, accountID string, repoUUID string) error {
+	var repoConfigs []models.RepositoryConfiguration
+	if err := r.db.
+		Where("org_id = ? and account_id = ? and repository_uuid = ?", orgID, accountID, repoUUID).
+		Find(&repoConfigs).
+		Error; err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r rpmDaoImpl) Create(orgID string, accountID string, repo *models.Repository, newRpm *models.Rpm) error {
+	// Check arguments
+	if orgID == "" {
+		return fmt.Errorf("orgID can not be an empty string")
+	}
+	if accountID == "" {
+		return fmt.Errorf("accountID can not be an empty string")
+	}
+	if repo == nil {
+		return fmt.Errorf("repo can not be nil")
+	}
+	if newRpm == nil {
+		return fmt.Errorf("newRpm can not be nil")
+	}
+
+	if err := r.isOwnedRepository(orgID, accountID, repo.UUID); err != nil {
+		return DBErrorToApi(err)
+	}
+
+	// Add Rpm record
+	if err := r.db.Create(newRpm).Error; err != nil {
+		return DBErrorToApi(err)
+	}
+
+	// Add at repositories_rpms the entry to relate to
+	var repositories_rpms []map[string]interface{} = []map[string]interface{}{
+		{
+			"repository_uuid": repo.UUID,
+			"rpm_uuid":        newRpm.UUID,
+		},
+	}
+	if err := r.db.Table(models.TableNameRpmsRepositories).Create(&repositories_rpms).Error; err != nil {
+		return DBErrorToApi(err)
+	}
+
+	return nil
+}
+
+func (r rpmDaoImpl) List(orgID string, accountID string, uuidRepo string, limit int, offset int) (api.RepositoryRpmCollectionResponse, int64, error) {
```

Comment:
only use orgID for authentication.  I kinda question if the dao layer here should care.  If the api layer has lookedup the uuid of the repo, maybe thats enough?

Reply:
i guess i'm saying maybe the authentication should be handled at the handler layer and not this layer